subprojects/docs/src/docs/userguide/authoring-builds/working_with_files.adoc

```diff
@@ -642,6 +642,30 @@ The `expand()` method treats the source files as https://docs.groovy-lang.org/la
 
 NOTE: It's good practice to specify the character set when reading and writing the file, otherwise the transformations won't work properly for non-ASCII text. You configure the character set with the link:{javadocPath}/org/gradle/api/file/CopySpec.html#setFilteringCharset-java.lang.String-[CopySpec.setFilteringCharset(String)] property. If it's not specified, the JVM default character set is used, which is likely to be different from the one you want.
 
+[[sec:setting_file_permissions]]
+=== Setting file permissions
+
+For any `CopySpec` involved in copying files, may it be the `Copy` task itself, or any child specifications, you can explicitly set the permissions the destination files will have, via the link:{javadocPath}/org/gradle/api/file/CopySpec.html#filePermissions-org.gradle.api.Action-[CopySpec.filePermissions {}] configurations block.
```

Comment:
The javadoc link in this paragraph currently points to the top of the `CopySpec`  page. The suggestion below makes it point directly to the `filePermissions {}` method but in the `CopyProcessingSpec` page. I think this is better, your call.

```suggestion
For any `CopySpec` involved in copying files, may it be the `Copy` task itself, or any child specifications, you can explicitly set the permissions the destination files will have, via the link:{javadocPath}/org/gradle/api/file/CopyProcessingSpec.html#filePermissions-org.gradle.api.Action-[CopySpec.filePermissions {}] configurations block.
```